docs/lib/views.md

```diff
@@ -1,5 +1,50 @@
-# ViewWithHeaders
+# AnsibleBaseView
 
-django-ansible-base provides a view `ansible_base.lib.utils.views.ViewWithHeaders` that is of type `rest_framework.views.APIView`. This view will look for the setting `ANSIBLE_BASE_EXTRA_HEADERS` which is a dict of key/value pairs (both strings). The view will inject headers in the format of `key: value` into the response.
+django-ansible-base provides a view called `ansible_base.lib.utils.views.AnsibleBaseView` which is indirectly parent class for all views in django-ansible-base.
 
-The view can be subclassed by any view created to inject default headers.
+This view itself is subclassed from `rest_framework.views.APIView` and is intended to be subclassed from individual services like:
+
+```
+from rest_framework.viewsets import ModelViewSet
+from ansible_base.lib.utils.views import AnsibleBaseView
+
+class MyServiceBaseApiView(AnsibleBaseView):
+    pass
+
+class MyServiceModelViewSet(ModelViewSet, MyServiceBaseApiView):
+    pass
+```
+
+## Changing the parent view
+
+All views in django-ansible-base actually inherit from a class called `ansible_base.lib.utils.views.AnsibleBaseDjanoAppApiView` which inherits, by default from `AnsibleBaseView`.
+
+However, if you already have an existing parent view with additional features you can override the view that `AnsibleBaseDjangoAppApiView` inherits from by setting the django setting `ANSIBLE_BASE_CUSTOM_VIEW_PARENT`.
```

Comment:
Yeah, I think AWX should use this. But when we do that, our base should inherit from `AnsibleBaseDjangoAppApiView`, just to add extra stuff.